compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2216,9 +2233,14 @@ public boolean isTypeCastable(BLangExpression expr, BType sourceType, BType targ
             return true;
         }
 
+
+        if (sourceType.tag == TypeTags.ANY && targetType.tag == TypeTags.READONLY) {
+            return true;
+        }
```

Comment:
Need to allow source - readonly and target - any too, right?

Reply:
Not sure. We cannot cast out error type. 
```Java
public boolean isTypeCastable(BLangExpression expr, BType sourceType, BType targetType, SymbolEnv env) {
        if (getTypeIntersection(sourceType, symTable.errorType, env) != symTable.semanticError
                && getTypeIntersection(targetType, symTable.errorType, env) == symTable.semanticError) {
            return false;
        }
```

This condition is false.

Reply:
Yes, my bad!

The following works.
```ballerina
public function main() {
    readonly x = 1;
    any y = <any> checkpanic x;
}
```